src/GrumPHP/Util/PhpVersion.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace GrumPHP\Util;
+
+/**
+ * Class PhpVersion
+ * @package GrumPHP\Util
+ */
+class PhpVersion
+{
+
+    /**
+     * @var array
+     */
+    private $versions;
+
+    /**
+     * PhpVersion constructor.
+     * @param array $versions
+     */
+    public function __construct(array $versions)
+    {
+        $this->versions = $versions;
+    }
+
+    /**
+     * Check if the current version is supported
+     * @param string $currentVersion
+     * @return bool
+     * @see https://secure.php.net/supported-versions.php for a list of currently supported versions
+     */
+    public function isSupportedVersion($currentVersion)
+    {
+        $versionIsSupported = false;
+        $now = new \DateTime();
+        foreach ($this->versions as $number => $eol) {
+            $eol = new \DateTime($eol);
+            if ($now < $eol && version_compare($currentVersion, $number) >= 0) {
+                $versionIsSupported = true;
```

Comment:
You could return it early here which will make the code a bit shorter and easier to follow.
Maybe you can also use $now <= $eol so that you can set the end date to the correct date with hours 00:00:00?